src/Nethermind/Nethermind.JsonRpc/Modules/Eth/Filter.cs

```diff
@@ -31,44 +32,65 @@ public void ReadJson(JsonElement filter, JsonSerializerOptions options)
                 filter = doc.RootElement;
             }
 
-            if (filter.TryGetProperty("blockHash"u8, out JsonElement blockHashElement))
-            {
-                blockHash = blockHashElement.GetString();
-            }
+            bool hasBlockHash = filter.TryGetProperty("blockHash"u8, out JsonElement blockHashElement);
+            bool hasFromBlock = filter.TryGetProperty("fromBlock"u8, out JsonElement fromBlockElement);
+            bool hasToBlock = filter.TryGetProperty("toBlock"u8, out JsonElement toBlockElement);
 
-            if (blockHash is null)
+            if (hasBlockHash)
             {
-                filter.TryGetProperty("fromBlock"u8, out JsonElement fromBlockElement);
-                FromBlock = BlockParameterConverter.GetBlockParameter(fromBlockElement.ToString());
-                filter.TryGetProperty("toBlock"u8, out JsonElement toBlockElement);
-                ToBlock = BlockParameterConverter.GetBlockParameter(toBlockElement.ToString());
+                if (hasFromBlock || hasToBlock)
+                {
+                    throw new ArgumentException("either (fromBlock and toBlock) or blockHash have to be specified");
+                }
+
+                FromBlock = new(new Hash256(blockHashElement.ToString()));
+                ToBlock = FromBlock;
             }
             else
             {
-                FromBlock = ToBlock = BlockParameterConverter.GetBlockParameter(blockHash);
+                FromBlock = hasFromBlock ? new BlockParameter(LongConverter.FromString(fromBlockElement.ToString())) : BlockParameter.Earliest;
```

Comment:
`BlockParameterConverter` also handled all non-integer values like `earliest`, `latest`, etc. I don't see this functionality in `LongConverter`, are these values still handled correctly?

In case if not - would be good to add some unit tests on these values specifically after a fix.

Reply:
My bad, they actually support it. Got confused